pkg/rule/retry/retryablerule.go

```diff
@@ -0,0 +1,98 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package retry
+
+import (
+	"context"
+	"log/slog"
+	"math"
+	"os"
+	"regexp"
+	"time"
+
+	"github.com/gardener/diki/pkg/rule"
+)
+
+type logger interface {
+	Info(string, ...any)
+}
+
+var _ rule.Rule = &RetryableRule{}
+
+// RetryableRule wraps [rule.Rule] and allows a rule to be retried when the retry condition is met.
+type RetryableRule struct {
+	BaseRule       rule.Rule
+	MaxRetries     int
+	RetryCondition func(ruleResult rule.RuleResult) bool
+	Logger         logger
+}
+
+// New creates a new RetryableRule.
+func New(options ...CreateOption) *RetryableRule {
+	handler := slog.NewJSONHandler(os.Stdout, &slog.HandlerOptions{Level: slog.LevelInfo})
```

Comment:
Shouldn't we discard the logs by default?
```suggestion
	slog.NewJSONHandler(io.Discard, nil)
```